posthog/settings/sentry.py

```diff
@@ -10,7 +10,39 @@
 from posthog.settings import get_from_env
 from posthog.settings.base_variables import TEST
 
-from datetime import datetime, timezone
+from dateutil import parser
+from random import random
+from datetime import timedelta
+
+
+def before_send_transaction(event, hint):
+    url_string = event.get("request", {}).get("url")
+    if url_string and "decide" in url_string:
+        DECIDE_SAMPLE_RATE = 0.00001  # 0.001%
+        should_sample = random() < DECIDE_SAMPLE_RATE
+
+        transaction_start_time = event.get("start_timestamp")
+        transaction_end_time = event.get("timestamp")
+        if transaction_start_time and transaction_end_time:
+            try:
+                parsed_start_time = parser.parse(transaction_start_time)
+                parsed_end_time = parser.parse(transaction_end_time)
+
+                duration = parsed_end_time - parsed_start_time
+
+                if duration >= timedelta(seconds=8):
+                    # return all events for transactions that took more than 8 seconds
+                    return event
+                elif duration > timedelta(seconds=2):
+                    # very high sample rate for transactions that took more than 2 seconds
+                    return event if random() < 0.5 else None
+
+            except Exception:
+                return event if should_sample else None
```

Comment:
nit picking that this could be a pass or a log and line 43 would still take effect...

Reply:
makes sense! but out of laziness / not wanting to wait for tests to pass again, I''l go ahead and merge, will fix this if this PR doesn't work 😂 🙈